nixos/server/default.nix

```diff
@@ -111,10 +111,56 @@
     '';
   };
 
-  # use TCP BBR has significantly increased throughput and reduced latency for connections
+  # https://www.kernel.org/doc/html/latest/networking/ip-sysctl.html
+  # In some cases, TCP BBR can significantly increase throughput and reduce latency,
+  # however this is not true in all cases, and should be used with caution
```

Comment:
What cases in particular?

Reply:
Are we talking about the long fat pipes issue?

Reply:
At edgecast, with ~20k machines covering something like >75% of routes in the world, we setup socket performance sampling using a a tool like xtcp2 (https://github.com/randomizedcoder/xtcp2).  This is essentially streaming back "ss --tcp --info", back to a big clickhouse cluster.  This gave us global visibility of socket performance.

Then we ran a series for "canaries"/experiments to enable BBR on some machines in different PoPs all over the world.

We then carefully analyzed the results to observed the impact to the socket performance.  In many cases, socket performance, like throughput, dropped.  This was particularly true for small HTTP transactions, and connections with low RTTs (like 20-40ms).  This might be because BBR takes time to find the target rate.  We did see benefits with BRR particularly to higher RTTs, and in particular cellular networks.  In the end, we kept cubic the default, and wrote automation to detect the sockets were BBR would benefit them, and then the Edgecast webserver would switch to BBR only for those destination routes.  This gave ~4% performance improvement globally.

Of course this was all for Edgecast, which has traffic patterns that could be very different to your use cases.

... I find that if I enable BBR on my laptop at home, it sucks, particularly for talking to local machines on my LAN, so I don't use BBR on my laptop anymore.  My internet connection is pretty good, so I'm ~12ms to most CDNs, so that's also why BBR isn't really required.

Anyway, my point is that just blindly turning on BBR could be doing more harm than good.

Reply:
Ok. So my assumption here, when I enabled `bbr` here, was that it would help for the typical server (1Gpbs-10Gpbs uplink, running in some data-center with reasonable global peering). That's why we also only put it in the "server" profile while keeping the default for the "desktop" profile.

But your data suggest that it would be actually worse for these cases. Interestingly your 4% performance improvement seem to match the result that Youtube reported as well. 

Which version of BBR did you use for your testing. It looks like since 2023 we now also have BBRv3, with some enhancements: https://datatracker.ietf.org/meeting/117/materials/slides-117-ccwg-bbrv3-algorithm-bug-fixes-and-public-internet-deployment-00